### PR TITLE
feat: wildcard for fetching all secrets

### LINKED ISSUE #234: How to import all keys from specific path
Is there a way to import all keys from specific path

### PR DESCRIPTION
**Pull request by matryxxx02**
fixes #234 

This PR allows the use of * (wildcard) to fetch all secrets into env variables.

# Example Usage  

example of secrets :
```json 
{
 ci : {
   aws : {
     URL_API : {
        "http://localhost:3000"
     }
   }
 }
}
```

```yaml
jobs:
    build:
        # ...
        steps:
            # ...
            - name: Import Secrets
              uses: hashicorp/vault-action@v2.3.1
              with:
                url: https://vault.mycompany.com:8200
                token: ${{ secrets.VaultToken }}
                caCertificate: ${{ secrets.VAULTCA }}
                secrets: |
                    secret/data/ci/aws *
            # ...
           - name : Test secret
              run: echo "Hello ${{env.URL_API}}"
```

### CODE REVIEW

**Comment by bfaulk96** [comment]
This would be great!

**Comment by bfaulk96** [comment]
@matryxxx02 any chance you could resolve those merge conflicts?

**Comment by bfaulk96** [comment]
Dang, that was fast!! @doowb What are the chances of this landing in the next release?

**Comment by ElenaForester** [comment]
Eagerly waiting for this to be released 🙏

**Comment by karlanuetzel** [comment]
+1 on this landing in next release

**Comment by ndrader** [comment]
Any news on this getting merged? Would love to use it.

**Comment by ElenaForester** [comment]
> Any news on this getting merged? Would love to use it.

I figured out that for now you can use 

> ```yaml
> secrets: |
>                     secret/data/ci/aws $.$
> ```

**Comment by Marcanta** [comment]
> > Any news on this getting merged? Would love to use it.
> 
> I figured out that for now you can use
> 
> > ```yaml
> > secrets: |
> >                     secret/data/ci/aws $.$
> > ```

I agree, however, it's annoying to use...

```yml
      - name: Import env.TEST from vault
        uses: hashicorp/vault-action@v2.3.0
        with:
          url: http://127.0.0.1:8200
          tlsSkipVerify: true
          token: ${{ secrets.VAULT_TOKEN }}
          secrets: |
              secret/data/test $.$ | TEST
              
      - name : echo env
        run: echo "Hello ${{env.TEST.HELLO_WORLD}}"
```

It would be great if we could drop the `TEST`.

**Comment by karlanuetzel** [comment]
Any update on this?

**Comment by hsimon-hashicorp** [comment]
> Currently going on `1 year, 47 days, 21 hours, 50 minutes` since this great PR was created and subsequently ignored.

You're right, and you have my apologies as well as my thanks for resurfacing this. I'll have our engineering team take a look. :)

**Comment by bfaulk96** [comment]
@matryxxx02 looks like there are once again a few merge conflicts, if you have time to address them

File: src/secrets.js [hunk 1]
```diff
@@ -46,16 +46,22 @@ async function getSecrets(secretRequests, client) {
...
             }
         }
-        if (!selector.match(/.*[\.].*/)) {
-            selector = '"' + selector + '"'
-        }
-        selector = "data." + selector
-        body = JSON.parse(body)
-        if (body.data["data"] != undefined) {
+
+        let value;
```
Comment (ltcarbonell) [inline src/secrets.js:50]: I believe we will need to pull out [this](https://github.com/hashicorp/vault-action/pull/247/files#diff-47be61e4446c15650db3c077e509c50ccfa2b4626c65fe43bad550fe25e611a9L53) JSON parse and move it here

**Review by ltcarbonell — CHANGES_REQUESTED** [review]
Good work so far, just a couple possible issues I noticed.  Thank you for sticking with this, I do think that this is a good feature we would want to include.

File: src/action.js [hunk 2]
```diff
@@ -8,6 +8,14 @@ const { auth: { retrieveToken }, secrets: { getSecrets } } = require('./index');
 const AUTH_METHODS = ['approle', 'token', 'github', 'jwt', 'kubernetes'];
 const ENCODING_TYPES = ['base64', 'hex', 'utf8'];
 
+function addMask(value) {
```
Comment (ltcarbonell) [inline src/action.js:11]: I think moving this to its own function is a good idea, however it seems to cause some issues with the acceptance tests

```
~$ npm run test

> vault-action@0.1.0 test
> jest

 PASS  src/auth.test.js
 FAIL  src/action.test.js
  ● exportSecrets › single-line secret gets masked

    expect(jest.fn()).toBeCalledTimes(expected)

    Expected number of calls: 1
    Received number of calls: 2

      334 |         await exportSecrets();
      335 |
    > 336 |         expect(command.issue).toBeCalledTimes(1);
          |                               ^
      337 |
      338 |         expect(command.issue).toBeCalledWith('add-mask', 'secret');
      339 |         expect(core.setOutput).toBeCalledWith('key', 'secret');

      at Object.toBeCalledTimes (src/action.test.js:336:31)

  ● exportSecrets › multi-line secret gets masked for each line

    expect(jest.fn()).toBeCalledTimes(expected)

    Expected number of calls: 2
    Received number of calls: 4

      354 |         await exportSecrets();
      355 |
    > 356 |         expect(command.issue).toBeCalledTimes(2); // 1 for each non-empty line.
          |                               ^
      357 |
      358 |         expect(command.issue).toBeCalledWith('add-mask', 'a multi-line string');
      359 |         expect(command.issue).toBeCalledWith('add-mask', 'with blank lines');

      at Object.toBeCalledTimes (src/action.test.js:356:31)

::add-mask::value
::add-mask::value
 PASS  src/retries.test.js

Test Suites: 1 failed, 2 passed, 3 total
Tests:       2 failed, 27 passed, 29 total
Snapshots:   0 total
Time:        2.532 s, estimated 3 s
Ran all test suites.
::add-mask::value
::add-mask::value
```

**Comment by maxcoulombe** [comment]
Hey everyone, thanks for showing interest in this feature. I'm happy to report that an equivalent implementation was recently merged in #488 . I'll close this PR for now as the functionality implemented appears equivalent.

The wildcard selector feature should be available in an up-coming v2.8.0 release, cheers!

**Pull request closed without merging**